datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -143,7 +143,10 @@ impl PhysicalExpr for ScalarFunctionExpr {
         // evaluate the function
         let output = match self.args.is_empty() {
             true => self.fun.invoke_no_args(batch.num_rows()),
-            false => self.fun.invoke(&inputs),
+            false => match self.fun.signature().volatility {
```

Comment:
If you modified invoke_batch as above, we could change this code to simply call `self.fun.invoke_batch()` always